### PR TITLE
implementation of path for describing state changes in a transition

### PR DESCRIPTION
**Pull request by f5io**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ The returned `Machine` will contain the following default methods:
 - `Machine.edge(to)` -> `string` - The `edge` method will return the name of the transition that fulfils the transtion to the supplied state, otherwise will throw an error if none is available
 - `Machine.will(...to)` -> `Boolean` - The `will` method takes any number of states and attempts to find a shortest path between the current state and each state supplied, eventually ending at the last supplied state, returning a `Boolean`
 - `Machine.thru(...to)` -> `Promise` - The `thru` method, similarly to the `will` method, takes any number of states and attempts to find a shortest path between the current state and each state supplied, eventually ending at the last supplied state, then enacts the change to the machine by transitioning through all the states
+- `Machine.path(...to)` -> `Boolean|Array<Array<String>>` - The `path` method, similarly to the `will` method takes any number of states and returns either `false` denoting an invalid transtion, or an `Array` of pair tuples, ie. `[ ['A', 'B'], ['B', 'C'] ]` denoting the state changes it would take to achieve the transition, not effecting any change on the machine
```
Comment (chriswessels) [inline README.md]: transition spelling

Comment (chriswessels) [reply README.md]: without effecting any change

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: node:latest
+      - image: node:8.8
```
Comment (chriswessels) [inline .circleci/config.yml]: 8.9?

Comment (f5io) [reply .circleci/config.yml]: unfortunately not, `npm login` is broken

File: src/index.js [hunk 3]
```diff
@@ -64,23 +64,28 @@ const createFinder = (edges) => ([ from, to ]) =>
 const prepareThru = (finder, allEdges, stateKey) => {
   const joins = getJoins(allEdges);
   const recurse = shortestPath(joins);
-  const getPaths = compose(map(x => recurse(...x)), getPairs);
-  const getNames = compose(map(x => finder(x).shift()), getPairs);
+ 
+  // if a path is found, get pairs of the sequence, ie. [ [ A, B ], [ B, C ] ]
+  const getPathPairs = x => !x.some(y => y === undefined)
```
Comment (badrequest400) [inline src/index.js]: clearer with an every y => y, rather than inverse logic

**Pull request merged**